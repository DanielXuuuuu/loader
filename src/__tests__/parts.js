const {
  selectPart,
  parseSFCParts,
} = require('../parts');
const { readFileSync } = require('fs');
const { join } = require('path');

describe('parts', () => {
  it('should parse rml', () => {
    const rml = readFileSync(join(__dirname, './parts.rml'), 'utf-8');
    const parts = parseSFCParts(rml);
    const script = selectPart(parts, 'script');
    expect(script).toMatchSnapshot();

    const template = selectPart(parts, 'template');
    expect(template).toMatchSnapshot();

    const style = selectPart(parts, 'style');
    expect(style).toMatchSnapshot();

    const importLinks = selectPart(parts, 'importLinks');
    expect(importLinks).toMatchSnapshot();
  });
});
