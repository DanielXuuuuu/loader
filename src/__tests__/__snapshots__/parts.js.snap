// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parts should parse rml 1`] = `
"import { useState, useCallback } from 'react';

 export default function (props) {
  const [foo, setFoo] = useState(true);
  const handleClick = useCallback((evt) => {
    setFoo(false);
  }, [foo]);

  return {
    color: 'yellow',
    handleClick,
    foo,
    fn: () => 'hello world',
  };
}"
`;

exports[`parts should parse rml 2`] = `
Array [
  "<div className=\\"container\\">
  <div>{ \\"插值\\" }</div>
  <div>{ foo }</div>
  <div>{ foo + '...' }</div>
  <div>{ foo ? '1' : '2' }</div>
  <div>{ fn(1) }</div>
  <div>
    <Button type=\\"primary\\" onClick={handleClick}>
      Primary Button
    </Button>
  </div>
</div>",
]
`;

exports[`parts should parse rml 3`] = `
".foo {
    color: red;
  }"
`;

exports[`parts should parse rml 4`] = `
Array [
  Object {
    "Button": "Button",
    "from": "@alifd/next",
  },
]
`;
